src/trails.erl

```diff
@@ -96,38 +110,51 @@ trail(PathMatch, ModuleHandler, Options, MetaData, Constraints) ->
    , constraints => Constraints
    }.
 
+%% @doc Gets the `path_match` from the given `trail` map.
 -spec path_match(map()) -> string() | binary().
 path_match(Trail) ->
  maps:get(path_match, Trail, []).
 
+%% @doc Gets the `handler` from the given `trail` map.
 -spec handler(map()) -> module().
 handler(Trail) ->
  maps:get(handler, Trail, []).
 
+%% @doc Gets the `options` from the given `trail` map.
  -spec options(map()) -> list().
 options(Trail) ->
  maps:get(options, Trail, []).
 
+%% @doc Gets the `metadata` from the given `trail` map.
  -spec metadata(map()) -> map().
 metadata(Trail) ->
  maps:get(metadata, Trail, []).
 
+%% @doc Gets the `constraints` from the given `trail` map.
```

Comment:
don't say it's a `map`. The type is opaque, so it should say "from the given `trail`"